### DISCUSSION TITLE
A site builder app with Next.js and Vercel

### DISCUSSION THREAD

**Discussion by OleksandrHubarEK** [vercel/community#1113] [Help] [answered]
Hi guys! 
I really like Vercel and what you are doing.

Now I am working on finding the right tools for the project. This is something like a site builder, where the user should have a separate environment in which he can build a site and then publish it. Since the symbiosis of Next.js + Vercel is very convenient, will it be possible to automatically create an environment for each user? I imagine it as a separate deployment using [CLI](https://vercel.com/docs/cli/deploying-from-cli) and passing the user ID through it. But looking at the [limitations](https://vercel.com/docs/concepts/limits/overview#domains), I'm afraid there will be scaling issues.

What do you think, maybe there are more convenient tools? And am I digging in the right direction?

**Comment by leerob** [comment, marked as answer]
You should check out [Platforms](https://github.com/vercel/platforms), which was built for this!

https://demo.vercel.pub/platforms-starter-kit
